runtime-modules/proposals/engine/src/lib.rs

```diff
@@ -381,7 +422,7 @@ decl_module! {
         }
 
         /// Cancel a proposal by its original proposer.
-        #[weight = 10_000_000] // TODO: adjust weight
+        #[weight = <T as Trait>::WeightInfo::cancel_proposal(T::MaxLocks::get())]
```

Comment:
[Here](https://github.com/paritytech/substrate/blob/v2.0.0/frame/babe/src/lib.rs#L266) I saw that they used this style since it's a similar case where `Trait` is bound by `pallet_timestamp` which has its own `WeightInfo` trait.

Reply:
Let me know if you still think it's better to introduce an alias such as `ProposalsEngineWeightInfo` to solve this. 

I tried using `type WeightInfo<T> = <T as Trait>::WeightInfo` and I can't because the name overlaps with the `trait WeightInfo`.

Reply:
I think the alias is a better option because you can use it also in the on_initialize() to simplify code.